### PR TITLE
Add some death tests to make sure we don't fail silently

### PR DESCRIPTION
**Pull request by wujingyue**
with unsupported logical-to-allocation transforms.

https://github.com/google/googletest/blob/main/docs/advanced.md#death-tests

### CODE REVIEW

File: csrc/tensor_metadata.cpp [hunk 1]
```diff
@@ -81,17 +81,14 @@ class ForwardTraverseFromLogicalToAlloc {
     auto [outer_size, outer_stride] = outer_it->second;
     NVF_ERROR(
         inner_stride * inner_size == outer_stride,
-        "The logical domain and allocation domain of fusion input/output ",
-        "tensors must be a one-to-one map, therefore, ",
```
Comment (wujingyue) [inline csrc/tensor_metadata.cpp:85]: I updated this comment -- we support beyond one-to-one but merging of discontiguous dimensions is still disallowed for a different reason.

**Comment by wujingyue** [comment]
!test

File: tests/cpp/test_allocation_domain.cpp [hunk 2]
```diff
@@ -1441,6 +1441,53 @@ TEST_F(AllocationDomainTest, InputAllocationIsSplit_Concrete) {
...
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({6});
+  TensorView* out = set(in);
+  fusion->addInput(in);
+  fusion->addOutput(out);
+
+  in->split(0, 2);
+  in->reorder({{1, 0}});
+  in->merge(0);
```
Comment (jjsjann123) [inline tests/cpp/test_allocation_domain.cpp:1479]: Not even a nitpick but just an FYI, the other merge api where you explicitly provide the inner/outer dimensions for the merge would also work. So you don't need the reorder above. i.e. in->merge(1, 0).

**Review by jjsjann123 — APPROVED** [review]
Thanks for updating the comment.

**Comment by wujingyue** [comment]
!test

**Pull request merged**